plugins/ingest/src/main/java/org/elasticsearch/ingest/Pipeline.java

```diff
@@ -82,7 +82,11 @@ public Pipeline create(String id, Map<String, Object> config, Map<String, Proces
                     for (Map.Entry<String, Map<String, Object>> entry : processor.entrySet()) {
                         Processor.Factory factory = processorRegistry.get(entry.getKey());
                         if (factory != null) {
-                            processors.add(factory.create(entry.getValue()));
+                            Map<String, Object> processorConfig = entry.getValue();
+                            processors.add(factory.create(processorConfig));
+                            if (processorConfig.isEmpty() == false) {
+                                throw new IllegalArgumentException("processor [" + entry.getKey() + "] has unused properties [" + Arrays.toString(processorConfig.keySet().toArray()) + "]");
```

Comment:
make the error a bit more understandable for users? Like "processor x doesn't support some of the provided configuration parameters" and list them like you do already...